### ISSUE TITLE
Content element selector doesn't let you select elements, only pages and articles

### ISSUE THREAD

**Issue by sietsevandreven** [contao/contao#2046]
**Affected version(s)**

Contao 4.9.2

**Description**

The content element selector in backend doesn't let you pick an element. Only lets you pick pages or articles.

**Comment by fritzmg** [comment]
The latest version is 4.9.4, did you verify it there?

**Comment by sietsevandreven** [comment]
The problem is a bit different in 4.9.4. In some cases the content element selector opens a popup backend. I then can browse the backend like normal, but unable to select a content ekement. In other cases the popup shows all elelments in the article I placed the content element selector in. Getting out of that article is no option.

**Comment by leofeyer** [comment]
How do we reproduce the issue in the [online demo](https://demo.contao.org/contao)?

**Comment by sietsevandreven** [comment]
Edit an article, doesn't matter which one... If you add a content element (Inhaltselemente) you are able to link to an element in whatever article... When you click on Change Selection a pop-up opens. In it the content of the article you are working in. If you want to choose an element of an other article you have to open the menu (in previous Contao versions you could just click on the back button and it would automatically open in article/element view only). When the menu opens, click on article and open a different article than you are working in. You then see all element in that article, but you can't select them.

**Comment by leofeyer** [comment]
I cannot reproduce this issue. Here is what I have done:

* Edit the article "Include Elements" in the online demo.
* Add a new content element and change the type to "Inhaltselement".
* Click on "Auswahl ändern", which opens the picker.
* Go to the articles module either via the tab menu or via the back button.
* Edit the article "Home" and pick a content element.
* Click on "Anwenden".

The element is selected without a problem.

**Comment by mandrael** [comment]
Same issue within "**Events**". There the workaround above is okay for admins. But the issue renders the picker useless for users. This can be reproduced in the online demo:
1) Go to https://demo.contao.org/contao?do=user
2) Enter James Wilson's user area
3) go to events and enter content area of any event, e.g. the first (Contao North-Day)
4) Create new element, "inhaltselement", "bezogenes inhaltselement": "auswahl ändern"

```
Error message:
Forbidden
Was ist das Problem?
Invalid content element ID 6.
```
The event has ID 6
Include Elemente / Article works, you can select an article. And the rest works, too.

PS: same issue in "News"/"Nachrichten"

**Comment by leofeyer** [comment]
I have debugged the issue and here is the problem: If `do=calendar`, Contao will use `tl_content_calendar::checkPermission()` instead of `tl_content::checkPermission()`. However, the picker URL is `contao/picker?context=dc.tl_content&extras%5BfieldType%5D=radio&extras%5Bsource%5D=tl_content.344&value=`, so the `do=calendar` part gets lost and Contao falls back to using `tl_content::checkPermission()`.

**Comment by leofeyer** [comment]
#2201 is in fact the same problem.

**Comment by ausi** [comment]
> #2201 is in fact the same problem.

The `CURRENT_ID` problem is a different one here, should be fixed by #3457
